### ISSUE TITLE
Crash when parsing malformed PNM file

### ISSUE THREAD

**Issue by retpoline** [CESNET/GPUJPEG#74]
Hi folks,

A crash was found while fuzz testing of the **gpujpeg** binary which can be triggered via a malformed PNM file.

```
echo -ne '\x33\x20' > crash.pnm

$ gpujpeg --encode crash.pnm test.jpg
GPUJPEG rev abd713d
CUDA driver version:   11.4
CUDA runtime version:  10.1
Using Device #0:       NVIDIA GeForce GTX 1650 (c.c. 7.5)
Image dimensions must be set to nonzero values!

Encoding Image [crash.pnm]: 0x0 (unknown) RGB
gpujpeg: GPUJPEG/src/gpujpeg_common.c:1071: gpujpeg_image_calculate_size: Assertion `0' failed.
Aborted (core dumped)
```

**Comment by MartinPulec** [comment]
Hi @retpoline,

thanks, I think that this should be now fixed with the commit 3f529890. I am afraid that there will be some further not handled error conditions, if you encounter something else, it would be great if you let us know.

**Comment by retpoline** [comment]
Great, thanks for the quick fix!
